doc/specs/#653 - Quake Mode/#653 - Quake Mode.md

```diff
@@ -0,0 +1,661 @@
+    it is.
+  - When a positive number is provided, the terminal will use that value as a
+    duration (in seconds) to slide the terminal into position when activated.
+  - The default would be some sensible value. The pane animation is .2s, so
+    `0.2` might be a resonable default here.
+
+We could have alternatively provided a `"dropdownSpeed"` setting, that provided
+a number of pixels per second. In my opinion, that would be harder for users to
+use correctly. I believe that it's easier for users to mentally picture "I'd
+like the dropdown to last 100ms" vs "My monitor is 1504px tall, so I need to set
+this to 15040 to make the window traverse the entire display in .1s"
+
+Some users might want to be able to use the global hotkey to hide the window
+when the window is already visible. This would let the hotkey act as a sort of
+global toggle for the Terminal window. Others might not like that behavior, and
+just want the action to always bring the Terminal into focus, and do nothing if
+the terminal is already focused. To facilitate both these use cases, we'll add
+the following property:
+
+* `"hideWhenVisible": bool`
+  - When `true`: (_default_) When this hotkey is pressed, and the terminal
+    window is currently active, minimize the window.
+      - When `dropdownDuration` is not `0`, then the window will slide back off
+        the top at the same speed as it would come down.
+  - When `false`: When this hotkey is pressed, and the terminal window is
+    currently active, do nothing.
+
+### Quake Mode
+
+In addition to just summoning the window from anywhere, some terminals also
+support a special "quake mode" buffer or window. This window is one that closely
+emulates the console from quake:
+* It's docked to the top of the screen
+* It takes the full width of the monitor, and only the bottom can be resized
+* It often doesn't have any other UI elements, like tabs
+
+For fun, we'll also be adding a special `"_quake"` window with the same
+behavior. If the user names a window `_quake`, then it will behave in the
+following special ways:
+
+* On launch, it will ignore the `initialPosition` and
+  `initialRows`/`initialCols` setting, and instead resize to the top half of the
+  monitor.
+* On launch, it will ignore the `launchMode` setting, and always launch in focus
+  mode.
+  - Users can disable focus mode on the `_quake` window if they do want tabs.
+* It will not be resizable from any side except the bottom of the window, nor
+  will it be drag-able.
+* It will not be a valid target for the "most recent window" for window
+  glomming. If it's the only open window, with `"windowingBehavior":
+  "useExisting*"`, then a new window will be created instead.
+  - It _is_ still a valid target for something like `wt -w _quake new-tab`
+
+A window at runtime can be renamed to become the `_quake` window (if no other
+`_quake` window exists). When it does, it will resize to the position of the
+quake window, and enter focus mode.

```

Comment:
What if I want to switch which window is my quake window? Like, if window 1 is quake, when I rename window 2 to quake, window 1 should automatically be renamed (I guess it should lose it's name?).

Reply:
It would also be nice if we could use an action/keybinding to promote the current window to be the quake window (or demote it to be a regular window).

Reply:
oh man if only there was some way to rename windows that was [already in progress](https://github.com/microsoft/terminal/compare/main...dev/migrie/f/rename-windows) 😜